dashboard/src/pages/TreeDetails/Tabs/Tests/IndividualTestsTable.tsx

```diff
@@ -151,57 +191,66 @@ export function IndividualTestsTable({
-        </TableHead>
+        </TH>
       );
     });
     // TODO: remove exhaustive-deps and change memo  (all tables)
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [groupHeaders, sorting]);
 
-  const modelRows = table.getRowModel().rows;
+  const { rows } = useMemo(() => {
+    return table.getRowModel();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [table, sorting]);
+
+  const parentRef = useRef<HTMLDivElement>(null);
+  const virtualizer = useVirtualizer({
+    count: rows.length,
+    estimateSize: () => ESTIMATED_ROW_HEIGHT,
+    getScrollElement: () => parentRef.current,
+    overscan: 5,
+  });
+  const virtualItems = virtualizer.getVirtualItems();
+
   const tableRows = useMemo((): JSX.Element[] => {
-    return modelRows?.length
-      ? modelRows.map(row => (
-          <TableRow
-            key={row.id}
-            data-state={row.getIsSelected() && 'selected'}
-            className="cursor-pointer hover:bg-lightBlue"
-          >
-            {row.getVisibleCells().map(cell => (
-              <TableCellWithLink
-                key={cell.id}
-                linkProps={{
-                  to: '/tree/$treeId/test/$testId',
-                  params: {
-                    testId: row.original.id,
-                  },
-                  search: s => s,
-                }}
-              >
-                {flexRender(cell.column.columnDef.cell, cell.getContext())}
-              </TableCellWithLink>
-            ))}
-          </TableRow>
-        ))
-      : [
-          <TableRow key="no-results">
-            <TableCell colSpan={columns.length} className="h-24 text-center">
-              <FormattedMessage id="global.noResults" />
-            </TableCell>
-          </TableRow>,
-        ];
-  }, [modelRows]);
+    return virtualItems.map(virtualRow => {
+      const row = rows[virtualRow.index] as Row<TIndividualTest>;
+      return <TableRowMemoized row={row} key={row.id} />;
+    });
+  }, [rows, virtualItems]);
+
+  const [paddingTop, paddingBottom] = useMemo(() => {
+    if (virtualItems.length === 0) return [0, 0];
+    return [
+      virtualItems[0].start,
+      virtualizer.getTotalSize() - virtualItems[virtualItems.length - 1].end,
+    ];
+  }, [virtualItems, virtualizer]);
 
   return (
-    <BaseTable headerComponents={tableHeaders}>
-      <TableBody>{...tableRows}</TableBody>
-    </BaseTable>
+    <div ref={parentRef} className="max-h-80 max-w-full overflow-auto">
+      <div className="rounded-lg border-x border-t border-darkGray bg-white text-sm text-black">
+        <table className="w-full">
+          <DumbTableHeader>{tableHeaders}</DumbTableHeader>
+          <TableBody>
+            <tr>
+              <td style={{ height: paddingTop }}></td>
+            </tr>
+            {tableRows}
+            <tr>
+              <td style={{ height: paddingBottom }}></td>
+            </tr>
```

Comment:
thy there is a `<tr>` with a single `td?`  I believe this is not valid HTML since rows should always have the same number of cells